### PR TITLE
Add a stream for push rule updates

### PR DESCRIPTION
**Pull request by NegativeMjark**
Adds a stream for the updates to per user push rules. So that they can be replicated to servers slaving from this server, and so that clients can be notified in /sync when the push rules change.

TODO:
* [x] Prod the notifier when the push rules change so that long poll requests wake up.
* [x] Include the push rules in the /sync response.

### CODE REVIEW

**Comment by erikjohnston** [comment]
retest this please

**Comment by NegativeMjark** [comment]
The ``m.push_rules`` is documented by matrix-org/matrix-doc#288

File: synapse/storage/__init__.py [hunk 1]
```diff
@@ -157,6 +160,11 @@ def __init__(self, db_conn, hs):
             prefilled_cache=presence_cache_prefill
         )
 
+        self.push_rules_stream_cache = StreamChangeCache(
+            "PushRulesStreamChangeCache",
+            self._push_rules_stream_id_gen.get_max_token()[0],
+        )
```
Comment (erikjohnston) [inline synapse/storage/__init__.py]: Might be worth prefilling?

Comment (NegativeMjark) [inline synapse/storage/__init__.py]: Done deda480 9848b54

**Comment by NegativeMjark** [comment]
@erikjohnston PTAL

**Comment by erikjohnston** [comment]
LGTM

**Pull request merged**